File: myblog-server/src/main/java/com/site/blog/service/TagService.java
package com.site.blog.service;

import com.site.blog.model.entity.Tag;
import com.baomidou.mybatisplus.extension.service.IService;
import com.site.blog.model.entity.BlogTagCount;

import java.util.List;

/**
 * <p>
 * 标签表 服务类
 * </p>
 *
 * @author: 南街
 * @since 2019-08-28
 */
public interface TagService extends IService<Tag> {

    List<BlogTagCount> getBlogTagCountForIndex();

    boolean clearTag(Integer tagId);

}

File: blog-sever/server-jpa/src/main/kotlin/com/site/blog/service/TagService.kt
package com.site.blog.service


import com.site.blog.model.entity.BlogTagCount
import com.site.blog.model.entity.Tag

/**
 *
 *
 * 标签表 服务类
 *
 *
 */
interface TagService  {
    /**
     * 让博客标签数指数
     *
     * @return [List]<[BlogTagCount]>
     */
    fun getBogTagCountForIndex(): List<BlogTagCount>

    /**
     * 清除标签
     *
     * @param tagId 标签id
     * @return boolean
     */
    fun clearTag(tagId: String?): Int
}